sale_commission/tests/test_sale_commission.py

```diff
@@ -692,3 +693,27 @@ def test_payment_date_settlement(self):
         )
         self.assertEqual(1, len(settlements))
         self.assertEqual(1, len(settlements.line_ids))
+
+    def test_partial_payment_amount_settlement(self):
+        sale_order = self._create_sale_order(
+            self.agent_monthly, self.partial_commission_paid
+        )
+        sale_order.action_confirm()
+        date = fields.Date.today()
+        invoice = self._invoice_sale_order(sale_order)
+        partial_amount = invoice.amount_total / 2
+        self.register_payment(invoice, date, amount=partial_amount)
+        self.assertTrue(invoice._get_reconciled_invoices_partials())
+        self._settle_agent(self.agent_monthly, 1)
+        settlements = self.env["sale.commission.settlement"].search(
+            [
+                (
+                    "agent_id",
+                    "=",
+                    self.agent_monthly.id,
+                ),
+                ("state", "=", "settled"),
+            ]
+        )
+        self.assertEqual(1, len(settlements))
+        self.assertEqual(1, len(settlements.line_ids))
```

Comment:
Could you add a check that there are now two agent lines in the invoice, and that one has `is_partial_settled == True` and the other `False`

Also, if you can do `partial_amount = invoice.amount_total / 4` or `/ 10` you could check the amounts as well (If you split in half, the amounts would be the same)